src/BottomNavigation/BottomNavigationItem.js

```diff
@@ -34,54 +40,50 @@ function getStyles(props, context) {
   return styles;
 }
 
-class BottomNavigationItem extends React.Component {
-  static propTypes = {
-    /**
-     * Set the icon representing the view for this item.
-     */
-    icon: React.PropTypes.node,
-    /**
-     * Set the label describing the view for this item.
-     */
-    label: React.PropTypes.node,
-    /**
-     * Override the inline-styles of the root element.
-     */
-    style: React.PropTypes.object,
-  };
+const BottomNavigationItem = (props, context) => {
+  const {
+    label,
+    icon,
+    style,
+    ...other,
+  } = props;
 
-  static contextTypes = {
-    muiTheme: React.PropTypes.object.isRequired,
-  };
+  const {prepareStyles} = context.muiTheme;
+  const styles = getStyles(props, context);
 
-  render() {
-    const {
-      label,
-      icon,
-      style,
-      ...other,
-    } = this.props;
+  const styledIcon = cloneElement(icon, {
+    style: Object.assign({}, styles.icon, icon.props.style),
+    color: icon.props.color || styles.iconColor,
```

Comment:
@jacksonrayhamilton You raised a good point. That's definitely a broader issue as we use the above pattern on all the codebase. So that something we could improve on the all codebase.